### ISSUE TITLE
Suggestion: Keep the GUI separate from the debloating tasks

### ISSUE THREAD

**Issue by glibg10b** [Sycnex/Windows10Debloater#390]
Clicking a button in the GUI freezes it until the current task finishes. This means I have to wait for it to finish before I can start the next task. I suggest running the debloating tasks in a separate process and queuing tasks when pressing buttons while the program is busy. 

I would recommend against starting a new process for every task, though. If too many tasks run in parallel, the system will lock up and start killing processes before they can complete.

**Comment by Sycnex** [comment]
Good point. I don’t have any experience with separating the processes from the main thread, but I imagine this could be done using Jobs. I’ll look into this. Thanks!

**Comment by glibg10b** [comment]
I discovered today that one can queue actions just by clicking the buttons while the GUI is frozen, so implementing this might not be necessary

**Comment by Sycnex** [comment]
I was considering creating Jobs though still, Since it can make the script run much faster, plus I can have additional logging and reporting.

**Comment by Zoran-Jankov** [comment]
You could use `Start-ThreadJob` cmdlet for creating separate threads.
